### PR TITLE
Add ability to request the 'product_name_languages' and 'ingredients_text_languages' fields

### PR DESCRIPTION
**Pull request by blazern**
A followup for https://github.com/openfoodfacts/openfoodfacts-dart/pull/185#issuecomment-882091255
This patch allows to request the `product_name_languages` and `ingredients_text_languages` fields, thus giving the ability to retrieve all available values for product's name and ingredients text.

### CODE REVIEW

File: lib/model/Product.dart [hunk 1]
```diff
@@ -313,47 +314,86 @@ class Product extends JsonObject {
...
+            _langFrom(key, ProductField.INGREDIENTS_TEXT_IN_LANGUAGES.key);
         if (lang != OpenFoodFactsLanguage.UNDEFINED) {
           result.ingredientsTextInLanguages ??= {};
           result.ingredientsTextInLanguages![lang] = json[key];
         }
       }
+      // The OFF API also can give values in all available languages for some
+      // of the fields. We store those values in same kinds of fields as
+      // the product_name_[2 letter language code] values.
+      if (key == ProductField.NAME_ALL_LANGUAGES.key) {
```
Comment (monsieurtanuki) [inline lib/model/Product.dart:370]: I think there's potential confusion here. Let me explain.
```dart
ProductField.NAME_IN_LANGUAGES: 'product_name_',
ProductField.NAME_ALL_LANGUAGES: 'product_name_languages',
// ...

if (key.startsWith(ProductField.NAME_IN_LANGUAGES.key)) {
// 'product_name_languages' matches, but is not a decent option
if (key == ProductField.NAME_ALL_LANGUAGES.key) {
// 'product_name_languages' does match AND is not a decent option
```

There are many `if else`s but your added code starts with a simple `if`, and after the `startsWith` tests.
In the case of `'product_name_languages'` a first `if` will be triggered BUT will always be irrelevant, and then "your" `if` will work.

My suggestion (in that file): always code with `else`, and start with "your" `== ProductField.NAME_ALL_LANGUAGES.key` test.

Something like that:
```dart
if (key == ProductField.NAME_ALL_LANGUAGES.key) {
} else if (key == ProductField.INGREDIENTS_TEXT_ALL_LANGUAGES.key) {
} else if (key.startsWith(ProductField.NAME_IN_LANGUAGES.key)) {
} else if (key.startsWith(ProductField.CATEGORIES_TAGS_IN_LANGUAGES.key)) {
} else if (key.startsWith(ProductField.INGREDIENTS_TAGS_IN_LANGUAGES.key)) {
} else if (key.startsWith(ProductField.LABELS_TAGS_IN_LANGUAGES.key)) {
} else if (key.startsWith(ProductField.COUNTRIES_TAGS_IN_LANGUAGES.key)) {
} else if (key.startsWith(ProductField.INGREDIENTS_TEXT_IN_LANGUAGES.key)) {
}
```

**Review by monsieurtanuki — CHANGES_REQUESTED** [review]
Hi @blazern!
Just a minor change requested to avoid confusion in the code.

File: lib/model/Product.dart (hunk 1 above)
Comment (blazern) [reply lib/model/Product.dart]: Done

**Review by monsieurtanuki — APPROVED** [review]
Thank you @blazern!

**Comment by blazern** [comment]
@stephanegigandet, could you please take a look? :)

**Review by stephanegigandet — APPROVED** [review]
Looks good to me, thank you!

**Comment by blazern** [comment]
Could someone merge the PR please? :)
I don't have the rights
@monsieurtanuki?

**Comment by blazern** [comment]
Thanks!

**Pull request merged**